src/device/socket/connectionmanager.rs

```diff
@@ -226,6 +226,18 @@ impl<H: Hal, T: Transport> VsockConnectionManager<H, T> {
         Ok(bytes_read)
     }
 
+    /// Returns whether the receive buffer is empty.
+    pub fn is_recv_buffer_empty(&mut self, peer: VsockAddr, src_port: u32) -> Result<bool> {
```

Comment:
I think an `available` method which returns the number of bytes available to read (as I was suggesting before) would be better, as it's useful for other things as well.

Reply:
Done